pkg/targetgroupbinding/resource_manager.go

```diff
@@ -209,15 +215,75 @@ func (m *defaultResourceManager) reconcileWithInstanceTargetType(ctx context.Con
 	return nil
 }
 
+func (m *defaultResourceManager) reconcileWithALBTargetType(ctx context.Context, tgb *elbv2api.TargetGroupBinding) error {
+	tgARN := tgb.Spec.TargetGroupARN
+	targets, err := m.targetsManager.ListTargets(ctx, tgARN, *tgb.Spec.TargetType)
+	if err != nil {
+		return err
+	}
+	// Target groups with alb target type can only have one target load balancer
+	if len(targets) == 1 {
+		return nil
+	}
+	loadBalancerARN, err := m.buildTargetLoadBalancerArn(ctx, tgb)
+	if err != nil {
+		return err
+	}
+	return m.targetsManager.RegisterTargets(ctx, tgARN, []elbv2sdk.TargetDescription{
+		{
+			Id:   awssdk.String(loadBalancerARN),
+			Port: awssdk.Int64(int64(tgb.Spec.IngressRef.Port.IntVal)),
+		},
+	})
+}
+
+func (m *defaultResourceManager) buildTargetLoadBalancerArn(ctx context.Context, tgb *elbv2api.TargetGroupBinding) (string, error) {
+	ingressName := tgb.Spec.IngressRef.Name
+	ingressKey := types.NamespacedName{Namespace: tgb.Namespace, Name: ingressName}
+	ingress := &v1beta1.Ingress{}
```

Comment:
I think this should use the v1 Ingress API. Earlier APIs are no longer supported in Kubernetes.

Reply:
Updated from networking v1beta1 to networking v1